packages/web3-eth-abi/src/api/logs_api.ts

```diff
@@ -20,6 +20,9 @@ import { decodeParameter, decodeParametersWith } from './parameters_api.js';
 
 const STATIC_TYPES = ['bool', 'string', 'int', 'uint', 'address', 'fixed', 'ufixed'];
 
+const _decodeParameter = (inputType: string, clonedTopic: string) =>
+	inputType === 'string' ? clonedTopic : decodeParameter(inputType, clonedTopic);
+
```

Comment:
When a indexed event input is of type string, we know that it's value is going to be the keccak256 hash of the original string value, so this change returns the hash instead of trying to decode it